src/utils/format.ts

```diff
@@ -39,5 +47,5 @@ export const fromUnitToDecimalBN = (balance: number | string, decimal: number):
 
 // truncate out of precision part
 export const truncateAmount = (amount: number | string, precision: number): number => {
-  return toBN(toBN(amount).toFormat(precision)).toNumber()
+  return toBN(toBN(amount).toFixed(precision)).toNumber()
```

Comment:
```suggestion
  return +toBN(amount).toFixed(precision)
```

easy way to cast string to number